File: algorithms/dynamicProgramming/FibonacciModified.java
import java.io.*;
import java.util.*;
import java.text.*;
import java.math.*;
import java.util.regex.*;

// Solution to https://www.hackerrank.com/challenges/fibonacci-modified

public class Solution {

    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        int A = in.nextInt();
        int B = in.nextInt();
        int N = in.nextInt();
        BigInteger[] arr = new BigInteger[N];
        arr[0] = BigInteger.valueOf(A);
        arr[1] = BigInteger.valueOf(B);
        for(int i = 2; i < N; i++){
            arr[i] = arr[i-2].add(arr[i-1].multiply(arr[i-1]));
        }
        System.out.println(arr[N-1]);
    }
}

File: algorithms/dynamicProgramming/fibonacci_modified.py
# Solution to https://www.hackerrank.com/challenges/fibonacci-modified

a = [int(x) for x in input().split()]
cache = [a[0], a[1]]
n = a[2]

for i in range(2, n):
    val = cache[i-2] + (cache[i-1] * cache[i-1])
    cache.append(val)

print(cache[n-1])
